v2/tools/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -0,0 +1,183 @@
+			scheme: scheme,
+		},
+		armID:  armID,
+		client: client,
+	}
+}
+
+// Name returns the ARM ID of the resource we're importing
+func (i *importableARMResource) Name() string {
+	return i.armID
+}
+
+// Resource returns the actual resource that is being imported.
+// Only available after the import is complete.
+func (i *importableARMResource) Resource() genruntime.MetaObject {
+	return i.resource
+}
+
+func (i *importableARMResource) Import(ctx context.Context) ([]ImportableResource, error) {
+	// Parse armID into a more useful form
+	id, err := arm.ParseResourceID(i.armID)
+	if err != nil {
+		return nil, err // arm.ParseResourceID already returns a good error, no need to wrap
+	}
+
+	// Create an importable blank object into which we capture the current state of the resource
+	importable, err := i.createImportableObjectFromID(id)
+	if err != nil {
+		// Error doesn't need additional context
+		return nil, err
+	}
+
+	// Get the current status of the object from ARM
+	status, err := i.getStatus(ctx, i.armID, importable)
+	if err != nil {
+		// Error doesn't need additional context
+		return nil, err
+	}
+
+	// Set up our object Spec
+	err = importable.InitializeSpec(status)
+	if err != nil {
+		return nil, errors.Wrapf(err, "setting status on Kubernetes resource for resource %s", i.armID)
+	}
+
+	i.resource = importable
+	return nil, nil
+}
+
+func (i *importableARMResource) createImportableObjectFromID(armID *arm.ResourceID) (genruntime.ImportableARMResource, error) {
+	gvk, err := i.groupVersionKindFromID(armID)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to determine GVK of resource")
+	}
+
+	obj, err := i.createBlankObjectFromGVK(gvk)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create blank resource")
+	}
+
+	importable, ok := obj.(genruntime.ImportableARMResource)
+	if !ok {
+		return nil, errors.Errorf(
+			"unable to create blank resource, expected %s to identify an importable ARM object", armID)
+	}
+
+	importable.SetName(i.nameFromID(armID))
+
+	return importable, nil
+}
+
+func (i *importableARMResource) getStatus(
+	ctx context.Context,
+	armID string,
+	importable genruntime.ImportableARMResource,
+) (genruntime.ConvertibleStatus, error) {
+	// Create an empty ARM status object into which we capture the current state of the resource
+	armStatus, err := genruntime.NewEmptyARMStatus(importable, i.scheme)
+	if err != nil {
+		return nil, errors.Wrapf(err, "constructing ARM status for resource: %q", armID)
+	}
+
+	// Call ARM to get the current state of the resource
+	_, err = i.client.GetByID(ctx, armID, importable.GetAPIVersion(), armStatus)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting status update from ARM for resource %s", armID)
+	}
+
+	// Convert the ARM shape to the Kube shape
+	status, err := genruntime.NewEmptyVersionedStatus(importable, i.scheme)
+	if err != nil {
+		return nil, errors.Wrapf(err, "constructing Kube status object for resource: %q", armID)
+	}
+
+	//TODO: Owner?
```

Comment:
Does this TODO need to be completed? I assume not in this PR because you don't do hierarchies here?

I guess it's a bit tricky because if you import a child resource (which pretty much everything except ResourceGroup is) you need owner set for it to actually be directly applyable

Reply:
To be handled in a different PR.